### PR TITLE
Management command to migrate user email domains

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
from @gherceg's original PR https://github.com/dimagi/commcare-hq/pull/30655
with some updates to reflect small changes in the codebase and make this command more re-usable.

It seems we might need to do this more than once based on a new partner request outlined here
https://dimagi-dev.atlassian.net/browse/SAAS-14702

so let's actually merge this command in and create some documentation around this workflow so that it's almost instantaneous next time. :)

### Safety story
only management command

### Automated test coverage
n/a

### QA Plan
QA was already done on the previous iteration

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/templates/users/email/deprecated_user.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    </ul>
+
+    The following data was not copied to your new user:
+    <ul>
+      <li>API keys</li>
+      <li>Pending CommCare invitations</li>
+    </ul>
+
+    The {{ old_username }} user is no longer active. You can log into your new user with the username
+    {{ new_username }}. If you are exempt from SSO, you can follow the <i>Forgot Password</i> workflow to successfully
```
Comment (millerdev) [inline corehq/apps/users/templates/users/email/deprecated_user.html:25]: [This comment](https://github.com/dimagi/commcare-hq/pull/30655#discussion_r759404415) (from original PR) said that this "SSO" language had been scrapped. Was this missed?

File: corehq/apps/users/templates/users/email/deprecated_user.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ - Project space memberships (including roles and locations)
+ - Saved reports
+ - Scheduled reports
+
+The following data was not copied to your new user:
+ - API keys
+ - Pending CommCare invitations
+
+The {{ old_username }} user is no longer active. You can log into your new user with the username
+{{ new_username }}. If you are exempt from SSO, you can follow the Forgot Password workflow to successfully login.
```
Comment (millerdev) [inline corehq/apps/users/templates/users/email/deprecated_user.txt:20]: Same as above "SSO" comment.

**Review by millerdev — APPROVED** [review]
I remember breathing a sigh of relief when the original PR was closed rather than merged. This command makes me nervous, but I have no hard evidence against it. I guess one data point we have is that it was used previously and did not have unexpected fallout?

File: corehq/apps/users/management/commands/clone_web_users.py [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+    help = """
+    `notify` command - sends an email to specified users explaining a planned migration will take place in 7 days
+    `run` command - creates new web users based on existing users, copying/transferring data like reports, domain
+    memberships, and feature flags. Passwords are reset, and 2FA and API keys are not copied over.
+    cchq --control <env> django-manage clone_web_users <command> <file.csv> --verbose --dry-run
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('command', choices=COMMAND_CHOICES)
+        parser.add_argument('file', help='')
```
Comment (AmitPhulera) [inline corehq/apps/users/management/commands/clone_web_users.py:54]: Can we mention what sort of files we support here?

File: corehq/apps/users/templates/users/email/notify_scheduled_user_migration.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    We have scheduled a migration to be run on {{ scheduled_date }} that will impact your CommCare user. Your organization has requested that your user, {{ old_username }}, be migrated to a new user with the username {{ new_username }}.
+  </p>
+  <p>
+    The following data will be copied to your new user:
+  </p>
+  <ul>
+    <li>General user data (name, phone number, email, etc.)</li>
+    <li>Project space memberships (including roles and locations)</li>
+    <li>Saved reports</li>
+    <li>Scheduled reports</li>
```
Comment (AmitPhulera) [inline corehq/apps/users/templates/users/email/notify_scheduled_user_migration.html:16]: Should we mention `exports` here and in other email templates?

**Review by AmitPhulera — APPROVED** [review]
Pair reviewed with @nospame. Looks good overall. 
Just a few nitpick on some documentation.

File: corehq/apps/users/templates/users/email/deprecated_user.html (hunk 1 above)
Comment (biyeun) [reply corehq/apps/users/templates/users/email/deprecated_user.html:25]: interesting, looks like it was never pushed to the branch from that original PR

**Pull request closed without merging**